### PR TITLE
Add onChange support

### PR DESCRIPTION
**Pull request by rdjpalmer**
Pass a function to `onChange`, which will notify you when the component becomes visible. Allows usage of `react-on-visible` in scenarios where using the `.visible` class isn't ideal, e.g., when using CSS modules

### CODE REVIEW

**Comment by dazld** [comment]
thanks for the PR! I'll review it locally and come back to you - first pass though, looks good.

File: src/index.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ class OnVisible extends Component {
             this.setState({
                 visible: true,
                 top
+            }, function() {
+                if (this.props.onChange) this.props.onChange(this.state.visible);
```
Comment (dazld) [inline src/index.js:40]: how about a default prop of a noop function? then wouldn't need the `if` here. also, at this point the visible state is always `true`. I've been thinking about an option for notifying on enter and leave (kind of a bounce), so perhaps it makes sense to leave the argument in for now for forward compatibility sake.

File: src/index.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ import cx from 'classnames';
 import debounce from './lib/debounce';
 
 class OnVisible extends Component {
-    constructor() {
-        super(...arguments);
+    constructor(props) {
+        super(props);
```
Comment (dazld) [inline src/index.js:8]: think should leave the super invocation with all arguments in - possibly subtle bugs might crawl in if not.

File: src/index.js (hunk 1 above)
Comment (dazld) [inline src/index.js]: could then remove the anonymous function too.

Comment (dazld) [reply src/index.js:40]: actually, this function isn't bound too, so needs to be a fat arrow to have access to `this`.

**Comment by dazld** [comment]
fixed these things up and merged. thanks for your contribution @rdjpalmer, published as `1.0.3`

File: src/index.js (hunk 2 above)
Comment (rdjpalmer) [reply src/index.js:8]: Is this actually the case? In the [React documentation](https://facebook.github.io/react/docs/reusable-components.html#setting-the-initial-state) they seem to only invoke `super` with the props argument. Curious as to why they'd do that if what you say is true

**Comment by rdjpalmer** [comment]
Excellent stuff, thanks @dazld

File: src/index.js (hunk 2 above)
Comment (dazld) [reply src/index.js:8]: I'm sure I saw somewhere `super(props, context)`, which was what I was thinking of. Maybe in the source.. I'll have to look.

**Pull request merged**